src/ralph/lib/metrics/middlewares.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import time
+
+from django.conf import settings
+from django.db import connection
+from django.utils.text import slugify
+from metrology import Metrology
+
+PROCESSING_TIME_METRIC_PREFIX = getattr(
+    settings, 'PROCESSING_TIME_METRIC_PREFIX', 'processing_time'
+)
+DB_QUERIES_COUNT_METRIC_PREFIX = getattr(
+    settings, 'DB_QUERIES_COUNT_METRIC_PREFIX', 'db_queries_count'
+)
+DB_QUERIES_TIME_METRIC_PREFIX = getattr(
+    settings, 'DB_QUERIES_TIME_METRIC_PREFIX', 'db_queries_time'
+)
+REQUESTS_COUNTER_METRIC_PREFIX = getattr(
+    settings, 'REQUESTS_COUNTER_METRIC_PREFIX', 'requests_count'
+)
+
+METRIC_NAME_TMPL = getattr(
+    settings,
+    'METRIC_NAME_TMPL',
+    '{prefix}.{url_name}.{request_method}.{status_code}'
+)
+
+logger = logging.getLogger(__name__)
+
+
+class RequestMetricsMiddleware(object):
+    """
+    Middleware reporting request metrics (such as processing time, database
+    queries count) to metrology.
+
+    How to use it:
+    * add this middleware at the beginning of MIDDLEWARE_CLASSES in your
+      settings
+    * define and start Metrology reporter in you Django settings (more details
+      here: https://metrology.readthedocs.io/en/latest/)
+    """
+
+    def process_request(self, request):
+        request._request_start_time = time.time()
```

Comment:
Better to use monotonic time to avoid a rare problem that may happen when request comes before a time change and the response is ready after it.